apis/k8s/v1alpha1/nodepool_types.go

```diff
@@ -304,6 +304,7 @@ type NodePoolObservation struct {
 	PublicIPs                []string `json:"publicIps,omitempty"`
 	AvailableUpgradeVersions []string `json:"availableUpgradeVersions,omitempty"`
 	CPUFamily                string   `json:"cpuFamily,omitempty"`
+	NodeCount                *int32   `json:"nodeCount,omitempty"`
```

Comment:
usually there are no pointers in these structs. can we use int32 instead?

Reply:
I used a pointer here for making semantic difference between unspecified value and zero-value. This semantic difference will help us to implement backward compatible solutions. So I am inclined to stick with using a pointer here for int32 type.  

Eg: Let's assume that somehow we fail or forget to update/replace the nodepool CRD in the cluster but new API is available in the codebase with zero value.